bindings/pyroot_experimental/PyROOT/test/tdirectory_attrsyntax.py

```diff
@@ -44,5 +44,13 @@ def test_readHisto_attrsyntax(self):
         self.checkHisto(self.dir0.dir1.h1)
         self.checkHisto(self.dir0.dir1.dir2.h2)
 
+    def test_caching(self):
+        # check that __dict__ of self.dir_caching is initially empty
+        self.assertFalse(self.dir0.__dict__)
+        self.dir0.h
+        # check that after call is not empty anymore
+        self.assertTrue(self.dir0.__dict__)
```

Comment:
I believe we also agreed to test that `__dict__` has only one entry and the value of that entry is equal to `self.dir0.h`. Same for the other tests below.